### PR TITLE
feat(application-plc): Parliamentary list creation backend

### PR DESCRIPTION
**Pull request by kksteini**
First draft of backend for the upcoming parliamentary signature application.
Currently behind a feature flag.

## Checklist:

- [X] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced the Parliamentary List Creation Module for managing parliamentary lists.
  - Added email template generation for application submission notifications.
  - Implemented new APIs for parliamentary candidacies and collections.

- **Improvements**
  - Enhanced forms for better handling of constituency and candidate data.
  - Updated error message handling and logging.

- **Bug Fixes**
  - Resolved issues with default values in forms and API responses.

- **Documentation**
  - Updated API specifications to reflect new endpoints and data structures.

### CODE REVIEW

File: libs/application/template-api-modules/src/lib/modules/templates/signature-collection/parliamentary-list-creation/parliamentary-list-creation.service.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            email: '',
+            areas: supervisor.constituency.map((constituency) => {
+              let [id, name] = constituency.split('|')
+              return {
+                areaId: id,
+              }
+            }),
+          })),
+        ),
+      collectionId: '',
```
Comment (kksteini) [inline libs/application/template-api-modules/src/lib/modules/templates/signature-collection/parliamentary-list-creation/parliamentary-list-creation.service.ts:79]:
```suggestion
      // TODO: determine collectionID
      collectionId: '',
```

File: libs/application/templates/signature-collection/parliamentary-list-creation/src/forms/Draft.ts [hunk 2]
```diff
@@ -397,7 +423,9 @@ export const Draft: Form = buildForm({
                       ' - ' +
                       formatNationalId(s.supervisor.nationalId) +
                       ' - ' +
-                      s.constituency,
+                      (s.constituency as unknown as string[])
+                        .map((e) => e.split('|')[1])
```
Comment (kksteini) [inline libs/application/templates/signature-collection/parliamentary-list-creation/src/forms/Draft.ts:427]:
```suggestion
                        .map((c) => c.split('|')[1])
```

File: libs/clients/signature-collection/src/lib/signature-collection-shared.service.ts [hunk 3]
```diff
@@ -77,6 +82,7 @@ export class SignatureCollectionSharedClientService {
       })
       return mapList(list, umbodList)
     } catch (error) {
+      console.log('Error fetching list', error)
```
Comment (kksteini) [inline libs/clients/signature-collection/src/lib/signature-collection-shared.service.ts:85]:
```suggestion
```

File: libs/clients/signature-collection/src/lib/signature-collection-shared.service.ts [hunk 4]
```diff
@@ -98,6 +103,16 @@ export class SignatureCollectionSharedClientService {
...
+    console.log('What are we collecting here?')
+    console.log({
+      requirementsMet,
+      canCreateInfo,
+      isPresidential,
+      isActive,
+      ownedLists,
+      areas,
+    })
+
```
Comment (kksteini) [inline libs/clients/signature-collection/src/lib/signature-collection-shared.service.ts:115]:
```suggestion
```

File: libs/clients/signature-collection/src/lib/signature-collection.service.ts [hunk 5]
```diff
@@ -134,6 +135,64 @@ export class SignatureCollectionClientService {
...
+        medmaelalistar: filteredAreas.map((area) => ({
+          svaediID: parseInt(area.id),
+          listiNafn: `${name} - ${area.name}`,
+        })),
+      },
+    })
+    if (filteredAreas.length !== candidacy.umbodList?.length) {
+      throw new Error('Not all lists created')
+    }
+    return { slug: 'frambodWowee' }
```
Comment (kksteini) [inline libs/clients/signature-collection/src/lib/signature-collection.service.ts:193]: Need to better determine what to return here

**Review by albinagu — APPROVED** [review]

**Review by obmagnusson — APPROVED** [review]

**Pull request merged**